diff --git a/include/velodyne_pointcloud/datacontainerbase.hpp b/include/velodyne_pointcloud/datacontainerbase.hpp
index 06cd621a..827a9006 100644
--- a/include/velodyne_pointcloud/datacontainerbase.hpp
+++ b/include/velodyne_pointcloud/datacontainerbase.hpp
@@ -33,12 +33,12 @@
 #ifndef VELODYNE_POINTCLOUD__DATACONTAINERBASE_HPP_
 #define VELODYNE_POINTCLOUD__DATACONTAINERBASE_HPP_
 
-#include <tf2/LinearMath/Quaternion.h>
-#include <tf2/LinearMath/Vector3.h>
-#include <tf2/buffer_core.h>
-#include <tf2/exceptions.h>
-#include <tf2_ros/buffer.h>
-#include <tf2_ros/transform_listener.h>
+#include <tf2/LinearMath/Quaternion.hpp>
+#include <tf2/LinearMath/Vector3.hpp>
+#include <tf2/buffer_core.hpp>
+#include <tf2/exceptions.hpp>
+#include <tf2_ros/buffer.hpp>
+#include <tf2_ros/transform_listener.hpp>
 
 #include <Eigen/Core>
 #include <Eigen/Geometry>
diff --git a/include/velodyne_pointcloud/organized_cloudXYZIRT.hpp b/include/velodyne_pointcloud/organized_cloudXYZIRT.hpp
index 87a286c7..0e95a609 100644
--- a/include/velodyne_pointcloud/organized_cloudXYZIRT.hpp
+++ b/include/velodyne_pointcloud/organized_cloudXYZIRT.hpp
@@ -33,7 +33,7 @@
 #ifndef VELODYNE_POINTCLOUD__ORGANIZED_CLOUDXYZIRT_HPP_
 #define VELODYNE_POINTCLOUD__ORGANIZED_CLOUDXYZIRT_HPP_
 
-#include <tf2/buffer_core.h>
+#include <tf2/buffer_core.hpp>
 
 #include <memory>
 #include <string>
diff --git a/include/velodyne_pointcloud/pointcloudXYZIRT.hpp b/include/velodyne_pointcloud/pointcloudXYZIRT.hpp
index 81fd9b2b..fbb2e3dd 100644
--- a/include/velodyne_pointcloud/pointcloudXYZIRT.hpp
+++ b/include/velodyne_pointcloud/pointcloudXYZIRT.hpp
@@ -33,7 +33,7 @@
 #ifndef VELODYNE_POINTCLOUD__POINTCLOUDXYZIRT_HPP_
 #define VELODYNE_POINTCLOUD__POINTCLOUDXYZIRT_HPP_
 
-#include <tf2/buffer_core.h>
+#include <tf2/buffer_core.hpp>
 
 #include <memory>
 #include <string>
diff --git a/include/velodyne_pointcloud/transform.hpp b/include/velodyne_pointcloud/transform.hpp
index 1f7ef7b0..38a37078 100644
--- a/include/velodyne_pointcloud/transform.hpp
+++ b/include/velodyne_pointcloud/transform.hpp
@@ -33,10 +33,10 @@
 #ifndef VELODYNE_POINTCLOUD__TRANSFORM_HPP_
 #define VELODYNE_POINTCLOUD__TRANSFORM_HPP_
 
-#include <message_filters/subscriber.h>
-#include <tf2_ros/buffer.h>
-#include <tf2_ros/message_filter.h>
-#include <tf2_ros/transform_listener.h>
+#include <message_filters/subscriber.hpp>
+#include <tf2_ros/buffer.hpp>
+#include <tf2_ros/message_filter.hpp>
+#include <tf2_ros/transform_listener.hpp>
 
 #include <memory>
 #include <string>
diff --git a/src/lib/organized_cloudXYZIRT.cpp b/src/lib/organized_cloudXYZIRT.cpp
index 429f80d4..6da53c8a 100644
--- a/src/lib/organized_cloudXYZIRT.cpp
+++ b/src/lib/organized_cloudXYZIRT.cpp
@@ -30,7 +30,7 @@
 // ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
-#include <tf2/buffer_core.h>
+#include <tf2/buffer_core.hpp>
 
 #include <cmath>
 #include <memory>
diff --git a/src/lib/pointcloudXYZIRT.cpp b/src/lib/pointcloudXYZIRT.cpp
index 25979c35..ab10300e 100644
--- a/src/lib/pointcloudXYZIRT.cpp
+++ b/src/lib/pointcloudXYZIRT.cpp
@@ -30,7 +30,7 @@
 // ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
-#include <tf2/buffer_core.h>
+#include <tf2/buffer_core.hpp>
 
 #include <memory>
 #include <string>
diff --git a/tests/test_row_step.cpp b/tests/test_row_step.cpp
index dc9df9ad..15a83c22 100644
--- a/tests/test_row_step.cpp
+++ b/tests/test_row_step.cpp
@@ -32,7 +32,7 @@
 
 #include <gtest/gtest.h>
 
-#include <tf2/buffer_core.h>
+#include <tf2/buffer_core.hpp>
 
 #include <memory>
 
