diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e08aa6..99f7010 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,23 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   tf2_ros
   trajectory_msgs
   visualization_msgs
+  interactive_markers
+)
+
+set(THIS_PACKAGE_INCLUDE_TARGET_DEPENDS
+  ${Boost_LIBRARIES}
+  ${geometry_msgs_TARGETS}
+  ${graph_msgs_TARGETS}
+  moveit_core::moveit_core
+  moveit_ros_planning::moveit_ros_planning
+  rclcpp::rclcpp
+  rviz_visual_tools::rviz_visual_tools
+  ${std_msgs_TARGETS}
+  tf2_eigen::tf2_eigen
+  tf2_ros::tf2_ros
+  ${trajectory_msgs_TARGETS}
+  ${visualization_msgs_TARGETS}
+  interactive_markers::interactive_markers
 )
 
 # Visualization Tools Library
@@ -46,7 +63,7 @@ target_include_directories(${PROJECT_NAME}
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   PUBLIC $<INSTALL_INTERFACE:include>
 )
-ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_TARGET_DEPENDS})
 
 # Demo executable
 add_executable(${PROJECT_NAME}_demo
@@ -55,7 +72,7 @@ add_executable(${PROJECT_NAME}_demo
 target_link_libraries(${PROJECT_NAME}_demo
   ${PROJECT_NAME}
 )
-ament_target_dependencies(${PROJECT_NAME}_demo ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(${PROJECT_NAME}_demo ${THIS_PACKAGE_INCLUDE_TARGET_DEPENDS})
 
 
 # Exports
