diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9140d2a4..043925338 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,8 +32,10 @@ target_include_directories(
 set_target_properties(
   moveit_robot_interaction PROPERTIES VERSION
                                       "${moveit_ros_robot_interaction_VERSION}")
-ament_target_dependencies(moveit_robot_interaction
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  moveit_robot_interaction moveit_core::moveit_core
+  moveit_ros_planning::moveit_ros_planning
+  interactive_markers::interactive_markers ${tf2_geometry_msgs_TARGETS})
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
@@ -41,8 +43,15 @@ if(BUILD_TESTING)
   target_link_libraries(locked_robot_state_test moveit_robot_interaction)
 endif()
 
+# Create an umbrella INTERFACE target so downstream packages can use
+# moveit_ros_robot_interaction::moveit_ros_robot_interaction to link all
+# libraries at once.
+add_library(moveit_ros_robot_interaction INTERFACE)
+target_link_libraries(moveit_ros_robot_interaction
+                      INTERFACE moveit_robot_interaction)
+
 install(
-  TARGETS moveit_robot_interaction
+  TARGETS moveit_ros_robot_interaction moveit_robot_interaction
   EXPORT moveit_ros_robot_interactionTargets
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
diff --git a/src/locked_robot_state.cpp b/src/locked_robot_state.cpp
index 15b8d0c84..35c859bda 100644
--- a/src/locked_robot_state.cpp
+++ b/src/locked_robot_state.cpp
@@ -66,7 +66,7 @@ void robot_interaction::LockedRobotState::setState(const moveit::core::RobotStat
 
     // If someone else has a reference to the state, then make a new copy.
     // The old state is orphaned (does not change, but is now out of date).
-    if (state_.unique())
+    if (state_.use_count() == 1)
     {
       *state_ = state;
     }
@@ -87,7 +87,7 @@ void robot_interaction::LockedRobotState::modifyState(const ModifyStateFunction&
 
     // If someone else has a reference to the state, then make a copy.
     // The old state is orphaned (does not change, but is now out of date).
-    if (!state_.unique())
+    if (state_.use_count() != 1)
       state_ = std::make_shared<moveit::core::RobotState>(*state_);
 
     modify(state_.get());
