diff --git a/CMakeLists.txt b/CMakeLists.txt
index b66f1b9..8bcb09b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,6 +42,7 @@ add_library(
   src/steering_kinematics.cpp
   src/steering_odometry.cpp
 )
+target_compile_definitions(steering_controllers_library PRIVATE _USE_MATH_DEFINES)
 target_include_directories(steering_controllers_library PUBLIC
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/steering_controllers_library>")
