diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2610a37ea..3d3515754 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@ find_package(urdf REQUIRED)
 set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
 
-qt5_wrap_cpp(
+qt6_wrap_cpp(
   MOC_FILES include/moveit_setup_core_plugins/start_screen_widget.hpp
   include/moveit_setup_core_plugins/configuration_files_widget.hpp
   include/moveit_setup_core_plugins/author_information_widget.hpp)
@@ -36,15 +36,15 @@ add_library(
 target_include_directories(
   ${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                          $<INSTALL_INTERFACE:include/moveit_setup_core_plugins>)
-ament_target_dependencies(
+target_link_libraries(
   ${PROJECT_NAME}
-  ament_index_cpp
-  moveit_ros_visualization
-  moveit_setup_framework
-  pluginlib
-  rclcpp
-  srdfdom
-  urdf)
+  ament_index_cpp::ament_index_cpp
+  moveit_ros_visualization::moveit_ros_visualization
+  moveit_setup_framework::moveit_setup_framework
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  srdfdom::srdfdom
+  urdf::urdf)
 
 install(
   TARGETS moveit_setup_core_plugins
diff --git a/src/configuration_files_widget.cpp b/src/configuration_files_widget.cpp
index 9e4c3484a..36d2ca8b3 100644
--- a/src/configuration_files_widget.cpp
+++ b/src/configuration_files_widget.cpp
@@ -44,7 +44,6 @@
 #include <QProgressBar>
 #include <QPushButton>
 #include <QPushButton>
-#include <QRegExp>
 #include <QSplitter>
 #include <QVBoxLayout>
 
