diff --git a/src/rclcpp/context.cpp b/src/rclcpp/context.cpp
index 0b472fd..8e022d5 100644
--- a/src/rclcpp/context.cpp
+++ b/src/rclcpp/context.cpp
@@ -425,7 +425,7 @@ Context::add_shutdown_callback(
   const ShutdownCallback & callback)
 {
   auto callback_shared_ptr =
-    std::make_shared<ShutdownCallbackHandle::ShutdownCallbackType>(callback);
+    std::make_shared<rclcpp::ShutdownCallbackHandle::ShutdownCallbackType>(callback);
 
   static_assert(
     shutdown_type == ShutdownType::pre_shutdown || shutdown_type == ShutdownType::on_shutdown);
@@ -438,7 +438,7 @@ Context::add_shutdown_callback(
     on_shutdown_callbacks_.emplace_back(callback_shared_ptr);
   }
 
-  ShutdownCallbackHandle callback_handle;
+  rclcpp::ShutdownCallbackHandle callback_handle;
   callback_handle.callback = callback_shared_ptr;
   return callback_handle;
 }
@@ -446,7 +446,7 @@ Context::add_shutdown_callback(
 template<Context::ShutdownType shutdown_type>
 bool
 Context::remove_shutdown_callback(
-  const ShutdownCallbackHandle & callback_handle)
+  const rclcpp::ShutdownCallbackHandle & callback_handle)
 {
   const auto callback_shared_ptr = callback_handle.callback.lock();
   if (callback_shared_ptr == nullptr) {
