diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb10aca60..7a3bcb819 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ find_package(rclcpp REQUIRED)
 set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
 
-qt5_wrap_cpp(MOC_FILES include/moveit_setup_app_plugins/launches_widget.hpp
+qt6_wrap_cpp(MOC_FILES include/moveit_setup_app_plugins/launches_widget.hpp
              include/moveit_setup_app_plugins/perception_widget.hpp)
 
 add_library(
@@ -33,9 +33,13 @@ target_include_directories(
   moveit_setup_app_plugins
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_setup_app_plugins>)
-ament_target_dependencies(
-  moveit_setup_app_plugins ament_index_cpp moveit_ros_visualization
-  moveit_setup_framework pluginlib rclcpp)
+target_link_libraries(
+  moveit_setup_app_plugins
+  ament_index_cpp::ament_index_cpp
+  moveit_ros_visualization::moveit_ros_visualization
+  moveit_setup_framework::moveit_setup_framework
+  pluginlib::pluginlib
+  rclcpp::rclcpp)
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
