diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95b504e..3ef2c1b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ include(CheckLibraryExists)
 
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros_core REQUIRED)
+find_package(Threads REQUIRED)
 
 if(UNIX AND NOT APPLE)
   include(cmake/check_c_compiler_uses_glibc.cmake)
@@ -90,6 +91,8 @@ target_link_libraries(${PROJECT_NAME}
   ${CMAKE_DL_LIBS}
   ament_cmake_ros_core::ament_ros_defaults
 )
+target_link_libraries(${PROJECT_NAME} Threads::Threads)
+ament_export_dependencies(Threads)
 
 check_library_exists(atomic __atomic_load_8 "" HAVE_LIBATOMICS)
 
diff --git a/src/testing/fault_injection.c b/src/testing/fault_injection.c
index a8b69d6..45fd5ae 100644
--- a/src/testing/fault_injection.c
+++ b/src/testing/fault_injection.c
@@ -16,7 +16,11 @@
 
 #include "rcutils/stdatomic_helper.h"
 
+#if defined(_WIN32) && !defined(__MINGW64__)
 static atomic_int_least64_t g_rcutils_fault_injection_count = {-1};
+#else
+static atomic_int_least64_t g_rcutils_fault_injection_count = -1;
+#endif
 
 void rcutils_fault_injection_set_count(int_least64_t count)
 {
