diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8dbdc3b..c2fba66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ find_package(rmw REQUIRED)
 find_package(rosgraph_monitor_msgs REQUIRED)
 find_package(rosidl_runtime_cpp REQUIRED)
 find_package(rosidl_typesupport_cpp REQUIRED)
+find_package(Threads REQUIRED)
 
 add_library(${PROJECT_NAME} SHARED
   src/shim.cpp
@@ -46,6 +47,7 @@ target_link_libraries(${PROJECT_NAME} PUBLIC
   rmw::rmw
   rosidl_runtime_cpp::rosidl_runtime_cpp
   rosidl_typesupport_cpp::rosidl_typesupport_cpp
+  Threads::Threads
   ${rosgraph_monitor_msgs_TARGETS}
 )
