diff --git a/mocap4r2_marker_viz/mocap4r2_marker_viz/CMakeLists.txt b/mocap4r2_marker_viz/mocap4r2_marker_viz/CMakeLists.txt
index ce11de5..2ad03ac 100644
--- a/mocap4r2_marker_viz/mocap4r2_marker_viz/CMakeLists.txt
+++ b/mocap4r2_marker_viz/mocap4r2_marker_viz/CMakeLists.txt
@@ -34,7 +34,13 @@ target_include_directories(${PROJECT_NAME}_NODE
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>)
 
-ament_target_dependencies(${PROJECT_NAME}_NODE rclcpp mocap4r2_msgs visualization_msgs geometry_msgs mocap4r2_marker_viz_srvs)
+target_link_libraries(${PROJECT_NAME}_NODE
+  rclcpp::rclcpp
+  ${mocap4r2_msgs_TARGETS}
+  ${visualization_msgs_TARGETS}
+  ${geometry_msgs_TARGETS}
+  ${mocap4r2_marker_viz_srvs_TARGETS}
+)
 
 target_link_libraries(mocap4r2_marker_viz ${PROJECT_NAME}_NODE)
 
