diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca30488f7..b6f5f2ad4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,19 +13,20 @@ find_package(eigen_stl_containers REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(eigen3_cmake_module REQUIRED)
 find_package(fcl REQUIRED)
+find_package(fmt REQUIRED)
 find_package(generate_parameter_library REQUIRED)
 find_package(geometric_shapes REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(kdl_parser REQUIRED)
 find_package(moveit_msgs REQUIRED)
-# Enforce system version liboctomap-dev
-# https://github.com/moveit/moveit2/issues/2862
-find_package(octomap 1.9.7...<1.10.0 REQUIRED)
+find_package(octomap REQUIRED)
 find_package(octomap_msgs REQUIRED)
+find_package(orocos_kdl REQUIRED)
 find_package(osqp REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(random_numbers REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rmw_implementation REQUIRED)
 find_package(rsl REQUIRED)
 find_package(ruckig REQUIRED)
 find_package(sensor_msgs REQUIRED)
@@ -68,8 +69,37 @@ add_subdirectory(transforms)
 add_subdirectory(utils)
 add_subdirectory(version)
 
+# Create an umbrella INTERFACE target so downstream packages can use
+# moveit_core::moveit_core to link all moveit_core libraries at once.
+add_library(moveit_core INTERFACE)
+target_link_libraries(
+  moveit_core
+  INTERFACE moveit_collision_detection
+            moveit_collision_detection_bullet
+            moveit_collision_detection_fcl
+            moveit_collision_distance_field
+            moveit_constraint_samplers
+            moveit_distance_field
+            moveit_dynamics_solver
+            moveit_exceptions
+            moveit_kinematic_constraints
+            moveit_kinematics_base
+            moveit_kinematics_metrics
+            moveit_macros
+            moveit_planning_interface
+            moveit_planning_scene
+            moveit_robot_model
+            moveit_robot_state
+            moveit_robot_trajectory
+            moveit_smoothing_base
+            moveit_test_utils
+            moveit_trajectory_processing
+            moveit_transforms
+            moveit_utils)
+
 install(
-  TARGETS moveit_collision_detection
+  TARGETS moveit_core
+          moveit_collision_detection
           moveit_collision_detection_bullet
           moveit_collision_detection_fcl
           moveit_collision_distance_field
@@ -119,6 +149,7 @@ ament_export_dependencies(
   Eigen3
   eigen3_cmake_module
   fcl
+  fmt
   generate_parameter_library
   geometric_shapes
   geometry_msgs
@@ -126,10 +157,12 @@ ament_export_dependencies(
   moveit_msgs
   octomap
   octomap_msgs
+  orocos_kdl
   osqp
   pluginlib
   random_numbers
   rclcpp
+  rmw_implementation
   rsl
   ruckig
   sensor_msgs
diff --git a/collision_detection/CMakeLists.txt b/collision_detection/CMakeLists.txt
index e3f8fd212..403c2af04 100644
--- a/collision_detection/CMakeLists.txt
+++ b/collision_detection/CMakeLists.txt
@@ -16,19 +16,19 @@ target_include_directories(
 include(GenerateExportHeader)
 generate_export_header(moveit_collision_detection)
 
-ament_target_dependencies(
+target_link_libraries(
   moveit_collision_detection
-  eigen_stl_containers
-  pluginlib
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  srdfdom
-  visualization_msgs
-  tf2_eigen
-  geometric_shapes
+  eigen_stl_containers::eigen_stl_containers
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  rmw_implementation::rmw_implementation
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  srdfdom::srdfdom
+  ${visualization_msgs_TARGETS}
+  tf2_eigen::tf2_eigen
+  geometric_shapes::geometric_shapes
   octomap)
 target_include_directories(
   moveit_collision_detection BEFORE
diff --git a/collision_detection/include/moveit/collision_detection/collision_common.hpp b/collision_detection/include/moveit/collision_detection/collision_common.hpp
index 19777a044..af3a89442 100644
--- a/collision_detection/include/moveit/collision_detection/collision_common.hpp
+++ b/collision_detection/include/moveit/collision_detection/collision_common.hpp
@@ -141,7 +141,33 @@ struct CostSource
   }
 };
 
-struct CollisionResult;
+/** \brief Representation of a collision checking result */
+struct CollisionResult
+{
+  using ContactMap = std::map<std::pair<std::string, std::string>, std::vector<Contact> >;
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  /** \brief Clear a previously stored result */
+  void clear()
+  {
+    collision = false;
+    distance = std::numeric_limits<double>::max();
+    contact_count = 0;
+    contacts.clear();
+    cost_sources.clear();
+  }
+  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
+  void print() const;
+  /** \brief True if collision was found, false otherwise */
+  bool collision = false;
+  /** \brief Closest distance between two bodies */
+  double distance = std::numeric_limits<double>::max();
+  /** \brief Number of contacts returned */
+  std::size_t contact_count = 0;
+  /** \brief A map returning the pairs of body ids in contact, plus their contact details */
+  ContactMap contacts;
+  /** \brief These are the individual cost sources when costs are computed */
+  std::set<CostSource> cost_sources;
+};
 
 /** \brief Representation of a collision checking request */
 struct CollisionRequest
@@ -159,9 +185,6 @@ struct CollisionRequest
   /** \brief If true, compute proximity distance */
   bool distance = false;
 
-  /** \brief If true, return detailed distance information. Distance must be set to true as well */
-  bool detailed_distance = false;
-
   /** \brief If true, a collision cost is computed */
   bool cost = false;
 
@@ -330,42 +353,4 @@ struct DistanceResult
   }
 };
 
-/** \brief Representation of a collision checking result */
-struct CollisionResult
-{
-  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
-
-  /** \brief Clear a previously stored result */
-  void clear()
-  {
-    collision = false;
-    distance = std::numeric_limits<double>::max();
-    distance_result.clear();
-    contact_count = 0;
-    contacts.clear();
-    cost_sources.clear();
-  }
-
-  /** \brief Throttled warning printing the first collision pair, if any. All collisions are logged at DEBUG level */
-  void print() const;
-
-  /** \brief True if collision was found, false otherwise */
-  bool collision = false;
-
-  /** \brief Closest distance between two bodies */
-  double distance = std::numeric_limits<double>::max();
-
-  /** \brief Distance data for each link */
-  DistanceResult distance_result;
-
-  /** \brief Number of contacts returned */
-  std::size_t contact_count = 0;
-
-  /** \brief A map returning the pairs of body ids in contact, plus their contact details */
-  using ContactMap = std::map<std::pair<std::string, std::string>, std::vector<Contact> >;
-  ContactMap contacts;
-
-  /** \brief These are the individual cost sources when costs are computed */
-  std::set<CostSource> cost_sources;
-};
 }  // namespace collision_detection
diff --git a/collision_detection/src/world.cpp b/collision_detection/src/world.cpp
index b2262f93d..8fd25b163 100644
--- a/collision_detection/src/world.cpp
+++ b/collision_detection/src/world.cpp
@@ -130,7 +130,7 @@ World::ObjectConstPtr World::getObject(const std::string& object_id) const
 
 void World::ensureUnique(ObjectPtr& obj)
 {
-  if (obj && !obj.unique())
+  if (obj && obj.use_count() != 1)
     obj = std::make_shared<Object>(*obj);
 }
 
diff --git a/collision_detection_bullet/CMakeLists.txt b/collision_detection_bullet/CMakeLists.txt
index feadeb653..1ea799949 100644
--- a/collision_detection_bullet/CMakeLists.txt
+++ b/collision_detection_bullet/CMakeLists.txt
@@ -18,35 +18,39 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(moveit_collision_detection_bullet
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_collision_detection_bullet SYSTEM BULLET)
-ament_target_dependencies(
+target_include_directories(moveit_collision_detection_bullet SYSTEM
+                           PUBLIC ${BULLET_INCLUDE_DIRS})
+target_link_libraries(
   moveit_collision_detection_bullet
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  octomap_msgs)
-target_link_libraries(moveit_collision_detection_bullet
-                      moveit_collision_detection moveit_utils)
+  moveit_collision_detection
+  moveit_utils
+  ${BULLET_LIBRARIES}
+  rclcpp::rclcpp
+  rmw_implementation::rmw_implementation
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS}
+  ${octomap_msgs_TARGETS})
 
 add_library(collision_detector_bullet_plugin SHARED
             src/collision_detector_bullet_plugin_loader.cpp)
 set_target_properties(collision_detector_bullet_plugin
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(collision_detector_bullet_plugin SYSTEM BULLET)
-ament_target_dependencies(
-  collision_detector_bullet_plugin
-  rclcpp
-  urdf
-  visualization_msgs
-  pluginlib
-  rmw_implementation
-  octomap_msgs)
+target_include_directories(collision_detector_bullet_plugin SYSTEM
+                           PUBLIC ${BULLET_INCLUDE_DIRS})
 target_link_libraries(
-  collision_detector_bullet_plugin moveit_collision_detection_bullet
-  moveit_planning_scene moveit_utils)
+  collision_detector_bullet_plugin
+  moveit_collision_detection_bullet
+  moveit_planning_scene
+  moveit_utils
+  ${BULLET_LIBRARIES}
+  rclcpp::rclcpp
+  urdf::urdf
+  ${visualization_msgs_TARGETS}
+  pluginlib::pluginlib
+  rmw_implementation::rmw_implementation
+  ${octomap_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
diff --git a/collision_detection_fcl/CMakeLists.txt b/collision_detection_fcl/CMakeLists.txt
index ece406480..63922f49e 100644
--- a/collision_detection_fcl/CMakeLists.txt
+++ b/collision_detection_fcl/CMakeLists.txt
@@ -11,26 +11,32 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(moveit_collision_detection_fcl
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
+target_link_libraries(
   moveit_collision_detection_fcl
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs)
-target_link_libraries(moveit_collision_detection_fcl moveit_collision_detection
-                      moveit_utils fcl)
+  moveit_collision_detection
+  moveit_utils
+  fcl
+  rclcpp::rclcpp
+  rmw_implementation::rmw_implementation
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS})
 
 add_library(collision_detector_fcl_plugin SHARED
             src/collision_detector_fcl_plugin_loader.cpp)
 set_target_properties(collision_detector_fcl_plugin
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(collision_detector_fcl_plugin rclcpp urdf
-                          visualization_msgs pluginlib rmw_implementation)
 target_link_libraries(
-  collision_detector_fcl_plugin moveit_collision_detection_fcl
-  moveit_planning_scene moveit_utils)
+  collision_detector_fcl_plugin
+  moveit_collision_detection_fcl
+  moveit_planning_scene
+  moveit_utils
+  rclcpp::rclcpp
+  urdf::urdf
+  ${visualization_msgs_TARGETS}
+  pluginlib::pluginlib
+  rmw_implementation::rmw_implementation)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
diff --git a/collision_detection_fcl/src/collision_common.cpp b/collision_detection_fcl/src/collision_common.cpp
index ca76202d6..34a4e1bd9 100644
--- a/collision_detection_fcl/src/collision_common.cpp
+++ b/collision_detection_fcl/src/collision_common.cpp
@@ -782,7 +782,7 @@ FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape,
         //        cache_it->second->collision_geometry_data_->getID().c_str());
         return cache_it->second;
       }
-      else if (cache_it->second.unique())
+      else if (cache_it->second.use_count() == 1)
       {
         const_cast<FCLGeometry*>(cache_it->second.get())->updateCollisionGeometryData(data, shape_index, false);
         //          RCLCPP_DEBUG(getLogger(), "Collision data structures for object %s retrieved from
@@ -806,7 +806,7 @@ FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape,
     auto cache_it = othercache.map_.find(wptr);
     if (cache_it != othercache.map_.end())
     {
-      if (cache_it->second.unique())
+      if (cache_it->second.use_count() == 1)
       {
         // remove from old cache
         FCLGeometryConstPtr obj_cache = cache_it->second;
@@ -839,7 +839,7 @@ FCLGeometryConstPtr createCollisionGeometry(const shapes::ShapeConstPtr& shape,
       auto cache_it = othercache.map_.find(wptr);
       if (cache_it != othercache.map_.end())
       {
-        if (cache_it->second.unique())
+        if (cache_it->second.use_count() == 1)
         {
           // remove from old cache
           FCLGeometryConstPtr obj_cache = cache_it->second;
diff --git a/collision_detection_fcl/src/collision_env_fcl.cpp b/collision_detection_fcl/src/collision_env_fcl.cpp
index b57e0767e..c655ed069 100644
--- a/collision_detection_fcl/src/collision_env_fcl.cpp
+++ b/collision_detection_fcl/src/collision_env_fcl.cpp
@@ -290,10 +290,6 @@ void CollisionEnvFCL::checkSelfCollisionHelper(const CollisionRequest& req, Coll
     dreq.enableGroup(getRobotModel());
     distanceSelf(dreq, dres, state);
     res.distance = dres.minimum_distance.distance;
-    if (req.detailed_distance)
-    {
-      res.distance_result = dres;
-    }
   }
 }
 
@@ -347,10 +343,6 @@ void CollisionEnvFCL::checkRobotCollisionHelper(const CollisionRequest& req, Col
     dreq.enableGroup(getRobotModel());
     distanceRobot(dreq, dres, state);
     res.distance = dres.minimum_distance.distance;
-    if (req.detailed_distance)
-    {
-      res.distance_result = dres;
-    }
   }
 }
 
diff --git a/collision_distance_field/CMakeLists.txt b/collision_distance_field/CMakeLists.txt
index 63b63a013..8bb5e3283 100644
--- a/collision_distance_field/CMakeLists.txt
+++ b/collision_distance_field/CMakeLists.txt
@@ -15,12 +15,17 @@ target_include_directories(
 set_target_properties(moveit_collision_distance_field
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_collision_distance_field urdf
-                          visualization_msgs tf2_eigen geometric_shapes octomap)
-
 target_link_libraries(
-  moveit_collision_distance_field moveit_planning_scene moveit_distance_field
-  moveit_collision_detection moveit_robot_state)
+  moveit_collision_distance_field
+  moveit_planning_scene
+  moveit_distance_field
+  moveit_collision_detection
+  moveit_robot_state
+  urdf::urdf
+  ${visualization_msgs_TARGETS}
+  tf2_eigen::tf2_eigen
+  geometric_shapes::geometric_shapes
+  octomap)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(
@@ -33,8 +38,6 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_collision_distance_field
                   test/test_collision_distance_field.cpp)
-  ament_target_dependencies(test_collision_distance_field geometric_shapes
-                            octomap srdfdom resource_retriever)
   target_link_libraries(
     test_collision_distance_field
     moveit_collision_distance_field
@@ -44,5 +47,9 @@ if(BUILD_TESTING)
     moveit_robot_state
     moveit_test_utils
     moveit_transforms
-    moveit_planning_scene)
+    moveit_planning_scene
+    geometric_shapes::geometric_shapes
+    octomap
+    srdfdom::srdfdom
+    resource_retriever::resource_retriever)
 endif()
diff --git a/constraint_samplers/CMakeLists.txt b/constraint_samplers/CMakeLists.txt
index 25cda7dc6..2cdbe05df 100644
--- a/constraint_samplers/CMakeLists.txt
+++ b/constraint_samplers/CMakeLists.txt
@@ -9,8 +9,6 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_constraint_samplers
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_constraint_samplers urdf urdfdom
-                          urdfdom_headers visualization_msgs)
 target_link_libraries(
   moveit_constraint_samplers
   moveit_robot_trajectory
@@ -18,7 +16,11 @@ target_link_libraries(
   moveit_kinematic_constraints
   moveit_kinematics_base
   moveit_planning_scene
-  moveit_utils)
+  moveit_utils
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
@@ -43,7 +45,6 @@ if(BUILD_TESTING)
                   test/pr2_arm_kinematics_plugin.cpp test/pr2_arm_ik.cpp)
   target_include_directories(test_constraint_samplers
                              PUBLIC ${geometry_msgs_INCLUDE_DIRS})
-  ament_target_dependencies(test_constraint_samplers kdl_parser)
   target_link_libraries(test_constraint_samplers moveit_test_utils moveit_utils
-                        moveit_constraint_samplers)
+                        moveit_constraint_samplers kdl_parser::kdl_parser)
 endif()
diff --git a/distance_field/CMakeLists.txt b/distance_field/CMakeLists.txt
index 165d82a7a..5c3723cc8 100644
--- a/distance_field/CMakeLists.txt
+++ b/distance_field/CMakeLists.txt
@@ -6,19 +6,21 @@ target_include_directories(
   moveit_distance_field
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_core>)
-target_link_libraries(moveit_distance_field moveit_macros moveit_utils)
-set_target_properties(moveit_distance_field
-                      PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
+target_link_libraries(
   moveit_distance_field
-  Boost
-  eigen_stl_containers
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  geometric_shapes
-  tf2_eigen
+  moveit_macros
+  moveit_utils
+  Boost::headers
+  Boost::iostreams
+  eigen_stl_containers::eigen_stl_containers
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS}
+  geometric_shapes::geometric_shapes
+  tf2_eigen::tf2_eigen
   octomap)
+set_target_properties(moveit_distance_field
+                      PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/dynamics_solver/CMakeLists.txt b/dynamics_solver/CMakeLists.txt
index 04afe66ef..c6c5a1d4e 100644
--- a/dynamics_solver/CMakeLists.txt
+++ b/dynamics_solver/CMakeLists.txt
@@ -6,8 +6,14 @@ target_include_directories(
 set_target_properties(moveit_dynamics_solver
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_dynamics_solver urdf urdfdom_headers
-                          orocos_kdl visualization_msgs kdl_parser)
-target_link_libraries(moveit_dynamics_solver moveit_robot_state moveit_utils)
+target_link_libraries(
+  moveit_dynamics_solver
+  moveit_robot_state
+  moveit_utils
+  urdf::urdf
+  urdfdom_headers::urdfdom_headers
+  ${orocos_kdl_LIBRARIES}
+  ${visualization_msgs_TARGETS}
+  kdl_parser::kdl_parser)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/exceptions/CMakeLists.txt b/exceptions/CMakeLists.txt
index 26591d955..04027c2ef 100644
--- a/exceptions/CMakeLists.txt
+++ b/exceptions/CMakeLists.txt
@@ -5,8 +5,8 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_exceptions PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_exceptions Boost rclcpp urdfdom
-                          urdfdom_headers)
-target_link_libraries(moveit_exceptions moveit_utils)
+target_link_libraries(
+  moveit_exceptions moveit_utils Boost::headers rclcpp::rclcpp
+  urdfdom::urdf_parser urdfdom_headers::urdfdom_headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/kinematic_constraints/CMakeLists.txt b/kinematic_constraints/CMakeLists.txt
index 51f33e495..534f51179 100644
--- a/kinematic_constraints/CMakeLists.txt
+++ b/kinematic_constraints/CMakeLists.txt
@@ -16,19 +16,20 @@ target_include_directories(
 set_target_properties(moveit_kinematic_constraints
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(
-  moveit_kinematic_constraints
-  urdf
-  urdfdom
-  urdfdom_headers
-  tf2_geometry_msgs
-  geometry_msgs
-  visualization_msgs
-  tf2_eigen)
-
 target_link_libraries(
-  moveit_kinematic_constraints moveit_collision_detection_fcl
-  moveit_kinematics_base moveit_robot_state moveit_robot_model moveit_utils)
+  moveit_kinematic_constraints
+  moveit_collision_detection_fcl
+  moveit_kinematics_base
+  moveit_robot_state
+  moveit_robot_model
+  moveit_utils
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${tf2_geometry_msgs_TARGETS}
+  ${geometry_msgs_TARGETS}
+  ${visualization_msgs_TARGETS}
+  tf2_eigen::tf2_eigen)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/kinematics_base/CMakeLists.txt b/kinematics_base/CMakeLists.txt
index 180253900..29ff3953e 100644
--- a/kinematics_base/CMakeLists.txt
+++ b/kinematics_base/CMakeLists.txt
@@ -11,16 +11,15 @@ target_include_directories(
   moveit_kinematics_base PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
 )# for this library
 
-# This line ensures that messages are built before the library is built
-ament_target_dependencies(
+target_link_libraries(
   moveit_kinematics_base
-  rclcpp
-  urdf
-  urdfdom_headers
-  srdfdom
-  moveit_msgs
-  geometric_shapes
-  geometry_msgs)
+  rclcpp::rclcpp
+  urdf::urdf
+  urdfdom_headers::urdfdom_headers
+  srdfdom::srdfdom
+  ${moveit_msgs_TARGETS}
+  geometric_shapes::geometric_shapes
+  ${geometry_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/moveit_kinematics_base_export.h
diff --git a/kinematics_metrics/CMakeLists.txt b/kinematics_metrics/CMakeLists.txt
index e760b834e..6255198ff 100644
--- a/kinematics_metrics/CMakeLists.txt
+++ b/kinematics_metrics/CMakeLists.txt
@@ -6,10 +6,13 @@ target_include_directories(
 set_target_properties(moveit_kinematics_metrics
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
-ament_target_dependencies(moveit_kinematics_metrics urdf urdfdom_headers
-                          visualization_msgs)
-
-target_link_libraries(moveit_kinematics_metrics moveit_robot_model
-                      moveit_robot_state moveit_utils)
+target_link_libraries(
+  moveit_kinematics_metrics
+  moveit_robot_model
+  moveit_robot_state
+  moveit_utils
+  urdf::urdf
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS})
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/online_signal_smoothing/CMakeLists.txt b/online_signal_smoothing/CMakeLists.txt
index 1f1912c6d..7d3b6dd01 100644
--- a/online_signal_smoothing/CMakeLists.txt
+++ b/online_signal_smoothing/CMakeLists.txt
@@ -5,12 +5,12 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
          $<INSTALL_INTERFACE:include/moveit_core>)
-target_link_libraries(moveit_smoothing_base ${Eigen_LIBRARIES} moveit_macros)
+target_link_libraries(moveit_smoothing_base ${Eigen_LIBRARIES} moveit_macros
+                      rclcpp::rclcpp tf2_eigen::tf2_eigen)
 include(GenerateExportHeader)
 generate_export_header(moveit_smoothing_base)
 set_target_properties(moveit_smoothing_base
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_smoothing_base rclcpp tf2_eigen)
 
 # Plugin implementations
 add_library(moveit_acceleration_filter SHARED src/acceleration_filter.cpp)
@@ -23,12 +23,13 @@ set_target_properties(moveit_acceleration_filter
 generate_parameter_library(moveit_acceleration_filter_parameters
                            src/acceleration_filter_parameters.yaml)
 target_link_libraries(
-  moveit_acceleration_filter moveit_acceleration_filter_parameters
-  moveit_robot_state moveit_smoothing_base osqp::osqp)
-ament_target_dependencies(
-  moveit_acceleration_filter srdfdom # include dependency from
-                                     # moveit_robot_model
-  pluginlib)
+  moveit_acceleration_filter
+  moveit_acceleration_filter_parameters
+  moveit_robot_state
+  moveit_smoothing_base
+  osqp::osqp
+  srdfdom::srdfdom
+  pluginlib::pluginlib)
 
 add_library(moveit_butterworth_filter SHARED src/butterworth_filter.cpp)
 generate_export_header(moveit_butterworth_filter)
@@ -41,11 +42,8 @@ generate_parameter_library(moveit_butterworth_filter_parameters
                            src/butterworth_parameters.yaml)
 target_link_libraries(
   moveit_butterworth_filter moveit_butterworth_filter_parameters
-  moveit_robot_model moveit_smoothing_base)
-ament_target_dependencies(
-  moveit_butterworth_filter
-  srdfdom # include dependency from moveit_robot_model
-  pluginlib)
+  moveit_robot_model moveit_smoothing_base srdfdom::srdfdom
+  pluginlib::pluginlib)
 
 add_library(moveit_ruckig_filter SHARED src/ruckig_filter.cpp)
 generate_export_header(moveit_ruckig_filter)
@@ -56,11 +54,13 @@ set_target_properties(moveit_ruckig_filter
 generate_parameter_library(moveit_ruckig_filter_parameters
                            src/ruckig_filter_parameters.yaml)
 target_link_libraries(
-  moveit_ruckig_filter moveit_robot_state moveit_ruckig_filter_parameters
-  moveit_smoothing_base ruckig::ruckig)
-ament_target_dependencies(
-  moveit_ruckig_filter srdfdom # include dependency from moveit_robot_model
-  pluginlib)
+  moveit_ruckig_filter
+  moveit_robot_state
+  moveit_ruckig_filter_parameters
+  moveit_smoothing_base
+  ruckig::ruckig
+  srdfdom::srdfdom
+  pluginlib::pluginlib)
 
 # Installation
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/planning_interface/CMakeLists.txt b/planning_interface/CMakeLists.txt
index 338765e2f..6a79dd7a0 100644
--- a/planning_interface/CMakeLists.txt
+++ b/planning_interface/CMakeLists.txt
@@ -6,9 +6,15 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_planning_interface
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_planning_interface moveit_msgs urdf urdfdom
-                          urdfdom_headers)
-target_link_libraries(moveit_planning_interface moveit_robot_trajectory
-                      moveit_robot_state moveit_planning_scene moveit_utils)
+target_link_libraries(
+  moveit_planning_interface
+  moveit_robot_trajectory
+  moveit_robot_state
+  moveit_planning_scene
+  moveit_utils
+  ${moveit_msgs_TARGETS}
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/planning_scene/CMakeLists.txt b/planning_scene/CMakeLists.txt
index debb3e5b6..c6eee289b 100644
--- a/planning_scene/CMakeLists.txt
+++ b/planning_scene/CMakeLists.txt
@@ -10,15 +10,6 @@ target_include_directories(
 # TODO: Fix the versioning
 set_target_properties(moveit_planning_scene
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_planning_scene
-  Boost
-  rclcpp
-  urdfdom
-  urdfdom_headers
-  octomap_msgs
-  octomap)
-
 target_link_libraries(
   moveit_planning_scene
   moveit_robot_model
@@ -30,7 +21,13 @@ target_link_libraries(
   moveit_kinematic_constraints
   moveit_robot_trajectory
   moveit_trajectory_processing
-  moveit_utils)
+  moveit_utils
+  Boost::headers
+  rclcpp::rclcpp
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${octomap_msgs_TARGETS}
+  octomap)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/moveit_planning_scene_export.h
@@ -47,9 +44,9 @@ if(BUILD_TESTING)
 
   ament_add_gtest(test_planning_scene test/test_planning_scene.cpp
                   APPEND_LIBRARY_DIRS "${APPEND_LIBRARY_DIRS}")
-  ament_target_dependencies(test_planning_scene geometric_shapes srdfdom)
-  target_link_libraries(test_planning_scene moveit_test_utils
-                        moveit_planning_scene)
+  target_link_libraries(
+    test_planning_scene moveit_test_utils moveit_planning_scene
+    geometric_shapes::geometric_shapes srdfdom::srdfdom)
 
   ament_add_gtest(test_collision_objects test/test_collision_objects.cpp
                   APPEND_LIBRARY_DIRS "${APPEND_LIBRARY_DIRS}")
diff --git a/robot_model/CMakeLists.txt b/robot_model/CMakeLists.txt
index 2886ed4f0..d6166a13d 100644
--- a/robot_model/CMakeLists.txt
+++ b/robot_model/CMakeLists.txt
@@ -28,25 +28,26 @@ target_include_directories(
     $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_model
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
+target_link_libraries(
   moveit_robot_model
-  angles
-  moveit_msgs
-  Eigen3
-  eigen_stl_containers
-  geometric_shapes
-  urdf
-  urdfdom_headers
-  srdfdom
-  visualization_msgs)
-target_link_libraries(moveit_robot_model moveit_exceptions moveit_macros
-                      moveit_utils)
+  moveit_exceptions
+  moveit_macros
+  moveit_utils
+  angles::angles
+  ${moveit_msgs_TARGETS}
+  Eigen3::Eigen
+  eigen_stl_containers::eigen_stl_containers
+  geometric_shapes::geometric_shapes
+  urdf::urdf
+  urdfdom_headers::urdfdom_headers
+  srdfdom::srdfdom
+  ${visualization_msgs_TARGETS})
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   ament_add_gtest(test_robot_model test/test.cpp)
-  ament_target_dependencies(test_robot_model rclcpp)
-  target_link_libraries(test_robot_model moveit_test_utils moveit_robot_model)
+  target_link_libraries(test_robot_model moveit_test_utils moveit_robot_model
+                        rclcpp::rclcpp)
 endif()
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
diff --git a/robot_state/CMakeLists.txt b/robot_state/CMakeLists.txt
index a906be706..44ccde09a 100644
--- a/robot_state/CMakeLists.txt
+++ b/robot_state/CMakeLists.txt
@@ -7,16 +7,17 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_state PROPERTIES VERSION
                                                     ${${PROJECT_NAME}_VERSION})
-ament_target_dependencies(
+target_link_libraries(
   moveit_robot_state
-  eigen_stl_containers
-  urdf
-  tf2_geometry_msgs
-  geometric_shapes
-  urdfdom_headers
-  Boost)
-target_link_libraries(moveit_robot_state moveit_robot_model
-                      moveit_kinematics_base moveit_transforms)
+  moveit_robot_model
+  moveit_kinematics_base
+  moveit_transforms
+  eigen_stl_containers::eigen_stl_containers
+  urdf::urdf
+  ${tf2_geometry_msgs_TARGETS}
+  geometric_shapes::geometric_shapes
+  urdfdom_headers::urdfdom_headers
+  Boost::headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
@@ -58,7 +59,7 @@ if(BUILD_TESTING)
 
   ament_add_google_benchmark(robot_state_benchmark
                              test/robot_state_benchmark.cpp)
-  ament_target_dependencies(robot_state_benchmark kdl_parser)
-  target_link_libraries(robot_state_benchmark moveit_robot_model
-                        moveit_test_utils moveit_robot_state)
+  target_link_libraries(
+    robot_state_benchmark moveit_robot_model moveit_test_utils
+    moveit_robot_state kdl_parser::kdl_parser)
 endif()
diff --git a/robot_state/src/attached_body.cpp b/robot_state/src/attached_body.cpp
index 3a606dfe5..04961a849 100644
--- a/robot_state/src/attached_body.cpp
+++ b/robot_state/src/attached_body.cpp
@@ -88,7 +88,7 @@ void AttachedBody::setScale(double scale)
   for (shapes::ShapeConstPtr& shape : shapes_)
   {
     // if this shape is only owned here (and because this is a non-const function), we can safely const-cast:
-    if (shape.unique())
+    if (shape.use_count() == 1)
     {
       const_cast<shapes::Shape*>(shape.get())->scale(scale);
     }
@@ -122,7 +122,7 @@ void AttachedBody::setPadding(double padding)
   for (shapes::ShapeConstPtr& shape : shapes_)
   {
     // if this shape is only owned here (and because this is a non-const function), we can safely const-cast:
-    if (shape.unique())
+    if (shape.use_count() == 1)
     {
       const_cast<shapes::Shape*>(shape.get())->padd(padding);
     }
diff --git a/robot_trajectory/CMakeLists.txt b/robot_trajectory/CMakeLists.txt
index c4b26ebb8..14594034b 100644
--- a/robot_trajectory/CMakeLists.txt
+++ b/robot_trajectory/CMakeLists.txt
@@ -5,11 +5,14 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_robot_trajectory
                       PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
-ament_target_dependencies(moveit_robot_trajectory rclcpp urdfdom
-                          urdfdom_headers)
-
-target_link_libraries(moveit_robot_trajectory moveit_robot_model
-                      moveit_robot_state moveit_utils)
+target_link_libraries(
+  moveit_robot_trajectory
+  moveit_robot_model
+  moveit_robot_state
+  moveit_utils
+  rclcpp::rclcpp
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/trajectory_processing/CMakeLists.txt b/trajectory_processing/CMakeLists.txt
index bbbe8d1f2..a4e3b128c 100644
--- a/trajectory_processing/CMakeLists.txt
+++ b/trajectory_processing/CMakeLists.txt
@@ -8,17 +8,18 @@ target_include_directories(
          $<INSTALL_INTERFACE:include/moveit_core>)
 set_target_properties(moveit_trajectory_processing
                       PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
+target_link_libraries(
   moveit_trajectory_processing
-  rclcpp
-  rmw_implementation
-  urdf
-  urdfdom
-  urdfdom_headers
-  visualization_msgs
-  Boost)
-target_link_libraries(moveit_trajectory_processing moveit_robot_state
-                      moveit_robot_trajectory ruckig::ruckig)
+  moveit_robot_state
+  moveit_robot_trajectory
+  ruckig::ruckig
+  rclcpp::rclcpp
+  rmw_implementation::rmw_implementation
+  urdf::urdf
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  ${visualization_msgs_TARGETS}
+  Boost::headers)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/transforms/CMakeLists.txt b/transforms/CMakeLists.txt
index 5880e5795..26316c2e8 100644
--- a/transforms/CMakeLists.txt
+++ b/transforms/CMakeLists.txt
@@ -3,18 +3,19 @@ target_include_directories(
   moveit_transforms
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_core>)
-target_link_libraries(moveit_transforms moveit_macros moveit_utils)
+target_link_libraries(
+  moveit_transforms
+  moveit_macros
+  moveit_utils
+  geometric_shapes::geometric_shapes
+  tf2_eigen::tf2_eigen
+  rclcpp::rclcpp
+  rmw_implementation::rmw_implementation
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  Boost::headers)
 set_target_properties(moveit_transforms PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(
-  moveit_transforms
-  geometric_shapes
-  tf2_eigen
-  rclcpp
-  rmw_implementation
-  urdfdom
-  urdfdom_headers
-  Boost)
 
 install(DIRECTORY include/ DESTINATION include/moveit_core)
 
diff --git a/utils/CMakeLists.txt b/utils/CMakeLists.txt
index ca44183fa..80dab7b2a 100644
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -3,8 +3,8 @@ add_library(moveit_utils SHARED src/lexical_casts.cpp src/message_checks.cpp
 target_include_directories(
   moveit_utils PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                       $<INSTALL_INTERFACE:include/moveit_core>)
-ament_target_dependencies(moveit_utils Boost moveit_msgs rclcpp fmt)
-target_link_libraries(moveit_utils rsl::rsl)
+target_link_libraries(moveit_utils rsl::rsl Boost::headers
+                      ${moveit_msgs_TARGETS} rclcpp::rclcpp fmt::fmt)
 set_target_properties(moveit_utils PROPERTIES VERSION
                                               "${${PROJECT_NAME}_VERSION}")
 
@@ -16,20 +16,21 @@ target_include_directories(
   moveit_test_utils
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_core>)
-target_link_libraries(moveit_test_utils moveit_robot_model
-                      moveit_kinematics_base rsl::rsl)
-ament_target_dependencies(
+target_link_libraries(
   moveit_test_utils
-  ament_index_cpp
-  Boost
-  geometry_msgs
-  urdf
-  pluginlib
-  srdfdom
-  urdfdom
-  urdfdom_headers
-  rclcpp
-  fmt)
+  moveit_robot_model
+  moveit_kinematics_base
+  rsl::rsl
+  ament_index_cpp::ament_index_cpp
+  Boost::headers
+  ${geometry_msgs_TARGETS}
+  urdf::urdf
+  pluginlib::pluginlib
+  srdfdom::srdfdom
+  urdfdom::urdf_parser
+  urdfdom_headers::urdfdom_headers
+  rclcpp::rclcpp
+  fmt::fmt)
 set_target_properties(moveit_test_utils PROPERTIES VERSION
                                                    "${${PROJECT_NAME}_VERSION}")
 
