diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,30 +64,98 @@
 )
 
 add_library(planning_context_loader_base SHARED src/planning_context_loader.cpp)
-target_link_libraries(planning_context_loader_base cartesian_limits_parameters)
-ament_target_dependencies(planning_context_loader_base
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  planning_context_loader_base
+  cartesian_limits_parameters
+  joint_limits_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(
   joint_limits_common SHARED
   src/joint_limits_aggregator.cpp src/joint_limits_container.cpp
   src/joint_limits_validator.cpp src/limits_container.cpp)
-target_link_libraries(joint_limits_common cartesian_limits_parameters)
-ament_target_dependencies(joint_limits_common ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  joint_limits_common
+  cartesian_limits_parameters
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(
   trajectory_generation_common SHARED
   src/trajectory_functions.cpp src/trajectory_generator.cpp
   src/trajectory_blender_transition_window.cpp)
-target_link_libraries(trajectory_generation_common cartesian_limits_parameters)
-ament_target_dependencies(trajectory_generation_common
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  trajectory_generation_common
+  cartesian_limits_parameters
+  joint_limits_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(command_list_manager SHARED src/command_list_manager.cpp
                                         src/plan_components_builder.cpp)
-target_link_libraries(command_list_manager trajectory_generation_common
-                      joint_limits_common)
-ament_target_dependencies(command_list_manager ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  command_list_manager
+  trajectory_generation_common
+  joint_limits_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 # ##############################################################################
 # Plugins ##
@@ -95,43 +163,128 @@
 
 add_library(pilz_industrial_motion_planner SHARED
             src/pilz_industrial_motion_planner.cpp)
-ament_target_dependencies(pilz_industrial_motion_planner
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(pilz_industrial_motion_planner
-                      planning_context_loader_base joint_limits_common)
+target_link_libraries(
+  pilz_industrial_motion_planner
+  planning_context_loader_base
+  joint_limits_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(
   planning_context_loader_ptp SHARED
   src/planning_context_loader_ptp.cpp src/trajectory_generator_ptp.cpp
   src/velocity_profile_atrap.cpp)
-ament_target_dependencies(planning_context_loader_ptp
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(planning_context_loader_ptp planning_context_loader_base
-                      joint_limits_common trajectory_generation_common)
+target_link_libraries(
+  planning_context_loader_ptp
+  planning_context_loader_base
+  joint_limits_common
+  trajectory_generation_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(
   planning_context_loader_lin SHARED
   src/planning_context_loader_lin.cpp src/trajectory_generator_lin.cpp
   src/velocity_profile_atrap.cpp)
-ament_target_dependencies(planning_context_loader_lin
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(planning_context_loader_lin planning_context_loader_base
-                      joint_limits_common trajectory_generation_common)
+target_link_libraries(
+  planning_context_loader_lin
+  planning_context_loader_base
+  joint_limits_common
+  trajectory_generation_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(
   planning_context_loader_circ SHARED
   src/planning_context_loader_circ.cpp src/trajectory_generator_circ.cpp
   src/path_circle_generator.cpp)
-ament_target_dependencies(planning_context_loader_circ
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(planning_context_loader_circ planning_context_loader_base
-                      joint_limits_common trajectory_generation_common)
+target_link_libraries(
+  planning_context_loader_circ
+  planning_context_loader_base
+  joint_limits_common
+  trajectory_generation_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 add_library(sequence_capability SHARED src/move_group_sequence_action.cpp
                                        src/move_group_sequence_service.cpp)
-ament_target_dependencies(sequence_capability ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(sequence_capability joint_limits_common
-                      command_list_manager trajectory_generation_common)
+target_link_libraries(
+  sequence_capability
+  joint_limits_common
+  command_list_manager
+  trajectory_generation_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 # ##############################################################################
 # Install ##
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -12,10 +12,26 @@
 
 # pilz_industrial_motion_testhelpers
 add_library(${PROJECT_NAME}_test_utils test_utils.cpp)
-ament_target_dependencies(${PROJECT_NAME}_test_utils
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(${PROJECT_NAME}_test_utils joint_limits_common
-                      trajectory_generation_common)
+target_link_libraries(
+  ${PROJECT_NAME}_test_utils
+  joint_limits_common
+  trajectory_generation_common
+  Eigen3::Eigen
+  ${geometry_msgs_TARGETS}
+  moveit_core::moveit_core
+  ${moveit_msgs_TARGETS}
+  moveit_ros_move_group::moveit_ros_move_group
+  moveit_ros_planning::moveit_ros_planning
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  tf2_eigen_kdl::tf2_eigen_kdl
+  ${tf2_geometry_msgs_TARGETS}
+  tf2_kdl::tf2_kdl
+  tf2_ros::tf2_ros
+  ${orocos_kdl_LIBRARIES}
+  Boost::headers)
 
 # Unit tests
 add_subdirectory(unit_tests)
diff --git a/test/unit_tests/CMakeLists.txt b/test/unit_tests/CMakeLists.txt
--- a/test/unit_tests/CMakeLists.txt
+++ b/test/unit_tests/CMakeLists.txt
@@ -38,9 +38,9 @@
 # Trajectory Functions Unit Test
 ament_add_gtest_executable(unittest_trajectory_functions
                            src/unittest_trajectory_functions.cpp)
-ament_target_dependencies(unittest_trajectory_functions Boost)
-target_link_libraries(unittest_trajectory_functions ${PROJECT_NAME}_test_utils
-                      trajectory_generation_common joint_limits_common)
+target_link_libraries(
+  unittest_trajectory_functions ${PROJECT_NAME}_test_utils
+  trajectory_generation_common joint_limits_common Boost::headers)
 add_ros_test(launch/unittest_trajectory_functions.test.py ARGS
              "test_binary_dir:=${CMAKE_CURRENT_BINARY_DIR}")
 
@@ -49,11 +49,12 @@
   unittest_trajectory_blender_transition_window
   src/unittest_trajectory_blender_transition_window.cpp
   ${CMAKE_SOURCE_DIR}/src/trajectory_blender_transition_window.cpp)
-ament_target_dependencies(unittest_trajectory_blender_transition_window
-                          pilz_industrial_motion_planner_testutils)
 target_link_libraries(
-  unittest_trajectory_blender_transition_window ${PROJECT_NAME}_test_utils
-  trajectory_generation_common planning_context_loader_lin)
+  unittest_trajectory_blender_transition_window
+  ${PROJECT_NAME}_test_utils
+  trajectory_generation_common
+  planning_context_loader_lin
+  pilz_industrial_motion_planner_testutils::pilz_industrial_motion_planner_testutils)
 add_ros_test(launch/unittest_trajectory_blender_transition_window.test.py ARGS
              "test_binary_dir:=${CMAKE_CURRENT_BINARY_DIR}")
 
@@ -70,20 +71,22 @@
 # trajectory generator circ Unit Test
 ament_add_gtest_executable(unittest_trajectory_generator_circ
                            src/unittest_trajectory_generator_circ.cpp)
-ament_target_dependencies(unittest_trajectory_generator_circ
-                          pilz_industrial_motion_planner_testutils)
-target_link_libraries(unittest_trajectory_generator_circ
-                      ${PROJECT_NAME}_test_utils planning_context_loader_circ)
+target_link_libraries(
+  unittest_trajectory_generator_circ
+  ${PROJECT_NAME}_test_utils
+  planning_context_loader_circ
+  pilz_industrial_motion_planner_testutils::pilz_industrial_motion_planner_testutils)
 add_ros_test(launch/unittest_trajectory_generator_circ.test.py ARGS
              "test_binary_dir:=${CMAKE_CURRENT_BINARY_DIR}")
 
 # trajectory generator lin Unit Test
 ament_add_gtest_executable(unittest_trajectory_generator_lin
                            src/unittest_trajectory_generator_lin.cpp)
-ament_target_dependencies(unittest_trajectory_generator_lin
-                          pilz_industrial_motion_planner_testutils)
-target_link_libraries(unittest_trajectory_generator_lin
-                      ${PROJECT_NAME}_test_utils planning_context_loader_lin)
+target_link_libraries(
+  unittest_trajectory_generator_lin
+  ${PROJECT_NAME}_test_utils
+  planning_context_loader_lin
+  pilz_industrial_motion_planner_testutils::pilz_industrial_motion_planner_testutils)
 add_ros_test(launch/unittest_trajectory_generator_lin.test.py ARGS
              "test_binary_dir:=${CMAKE_CURRENT_BINARY_DIR}")
 
