diff --git a/src/gz_ros2_control_plugin.cpp b/src/gz_ros2_control_plugin.cpp
index 236be25c..bdbc85da 100644
--- a/src/gz_ros2_control_plugin.cpp
+++ b/src/gz_ros2_control_plugin.cpp
@@ -214,7 +214,7 @@ GazeboSimROS2ControlPluginPrivate::GetEnabledJoints(
       case sdf::JointType::FIXED:
         {
           RCLCPP_INFO(
-            node_->get_logger(), "[gz_ros2_control] Fixed joint ['%s'] (Entity='%lu') is skipped.",
+            node_->get_logger(), "[gz_ros2_control] Fixed joint ['%s'] (Entity='%llu') is skipped.",
             jointName.c_str(), jointEntity);
           continue;
         }
@@ -225,7 +225,7 @@ GazeboSimROS2ControlPluginPrivate::GetEnabledJoints(
         {
           RCLCPP_WARN(
             node_->get_logger(),
-            "[gz_ros2_control] Joint ['%s'] (Entity='%lu') is of unsupported type."
+            "[gz_ros2_control] Joint ['%s'] (Entity='%llu') is of unsupported type."
             " Only joints with a single axis are supported.",
             jointName.c_str(), jointEntity);
           continue;
