diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,13 +22,23 @@
                                           src/chomp_planning_context.cpp)
 set_target_properties(moveit_chomp_interface
                       PROPERTIES VERSION "${moveit_planners_chomp_VERSION}")
-ament_target_dependencies(moveit_chomp_interface
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  moveit_chomp_interface
+  moveit_core::moveit_core
+  chomp_motion_planner::chomp_motion_planner
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  rsl::rsl)
 
 add_library(moveit_chomp_planner_plugin SHARED src/chomp_plugin.cpp)
-ament_target_dependencies(moveit_chomp_planner_plugin
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(moveit_chomp_planner_plugin moveit_chomp_interface)
+target_link_libraries(
+  moveit_chomp_planner_plugin
+  moveit_chomp_interface
+  moveit_core::moveit_core
+  chomp_motion_planner::chomp_motion_planner
+  pluginlib::pluginlib
+  rclcpp::rclcpp
+  rsl::rsl)
 
 install(
   TARGETS moveit_chomp_interface moveit_chomp_planner_plugin
