diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5febf7d..308eca6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,11 +188,11 @@ endif(roscpp_FOUND)
 #Ros2#
 if(rclcpp_FOUND)
 
-  ament_target_dependencies(rslidar_sdk_node 
-    rclcpp 
-    std_msgs 
-    sensor_msgs 
-    rslidar_msg)
+  target_link_libraries(rslidar_sdk_node 
+    rclcpp::rclcpp 
+    ${std_msgs_TARGETS}
+    ${sensor_msgs_TARGETS}
+    ${rslidar_msg_TARGETS})
 
   install(TARGETS
     rslidar_sdk_node
