diff --git a/CMakeLists.txt b/CMakeLists.txt
index 955f227d87..d5d828caf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,10 +22,15 @@ include_directories(include)
 # Planner Plugin
 add_library(stomp_moveit_plugin SHARED src/stomp_moveit_planner_plugin.cpp
                                        src/stomp_moveit_planning_context.cpp)
-ament_target_dependencies(stomp_moveit_plugin moveit_core std_msgs tf2_eigen
-                          visualization_msgs)
-target_link_libraries(stomp_moveit_plugin stomp::stomp stomp_moveit_parameters
-                      rsl::rsl)
+target_link_libraries(
+  stomp_moveit_plugin
+  stomp::stomp
+  stomp_moveit_parameters
+  rsl::rsl
+  moveit_core::moveit_core
+  ${std_msgs_TARGETS}
+  tf2_eigen::tf2_eigen
+  ${visualization_msgs_TARGETS})
 
 pluginlib_export_plugin_description_file(moveit_core
                                          stomp_moveit_plugin_description.xml)
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 3f83ea5a76..95b69c2650 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,8 +1,8 @@
 find_package(ament_cmake_gtest REQUIRED)
 ament_add_gtest(test_noise_generator test_noise_generator.cpp)
-ament_target_dependencies(test_noise_generator tf2_eigen)
-target_link_libraries(test_noise_generator stomp::stomp rsl::rsl)
+target_link_libraries(test_noise_generator stomp::stomp rsl::rsl
+                      tf2_eigen::tf2_eigen)
 
 ament_add_gtest(test_cost_functions test_cost_functions.cpp)
-ament_target_dependencies(test_cost_functions moveit_core)
-target_link_libraries(test_cost_functions stomp::stomp rsl::rsl)
+target_link_libraries(test_cost_functions stomp::stomp rsl::rsl
+                      moveit_core::moveit_core)
