diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e34b6346..2820d3390 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,7 +10,7 @@ find_package(ament_cmake_ros REQUIRED)
 find_package(moveit_setup_framework REQUIRED)
 find_package(pluginlib REQUIRED)
 
-qt5_wrap_cpp(
+qt6_wrap_cpp(
   MOC_FILES
   include/moveit_setup_srdf_plugins/collision_linear_model.hpp
   include/moveit_setup_srdf_plugins/collision_matrix_model.hpp
@@ -50,8 +50,9 @@ target_include_directories(
   moveit_setup_srdf_plugins
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_setup_srdf_plugins>)
-ament_target_dependencies(moveit_setup_srdf_plugins moveit_setup_framework
-                          pluginlib)
+target_link_libraries(
+  moveit_setup_srdf_plugins moveit_setup_framework::moveit_setup_framework
+  pluginlib::pluginlib)
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
diff --git a/include/moveit_setup_srdf_plugins/collision_matrix_model.hpp b/include/moveit_setup_srdf_plugins/collision_matrix_model.hpp
index 57e34ef04..296a6d8eb 100644
--- a/include/moveit_setup_srdf_plugins/collision_matrix_model.hpp
+++ b/include/moveit_setup_srdf_plugins/collision_matrix_model.hpp
@@ -65,7 +65,7 @@ public:
   void setEnabled(const QModelIndexList& indexes, bool value);
 
 public Q_SLOTS:
-  void setFilterRegExp(const QString& filter);
+  void setFilterRegularExpression(const QString& filter);
 
 private:
   LinkPairMap::iterator item(const QModelIndex& index);
diff --git a/src/collision_linear_model.cpp b/src/collision_linear_model.cpp
index 9fb5b65d1..cca8a6aeb 100644
--- a/src/collision_linear_model.cpp
+++ b/src/collision_linear_model.cpp
@@ -39,6 +39,7 @@
 
 #include <QItemSelection>
 #include <QPainter>
+#include <QRegularExpression>
 #include <cmath>
 namespace moveit_setup
 {
@@ -264,7 +265,7 @@ bool SortFilterProxyModel::filterAcceptsRow(int source_row, const QModelIndex& s
       m->data(m->index(source_row, 2), Qt::CheckStateRole) != Qt::Checked)
     return false;  // not accepted due to check state
 
-  const QRegExp regexp = filterRegExp();
+  const QRegularExpression regexp = filterRegularExpression();
-  if (regexp.isEmpty())
+  if (regexp.pattern().isEmpty())
   return true;
 
diff --git a/src/collision_matrix_model.cpp b/src/collision_matrix_model.cpp
index f8de45156..6f06ee2f2 100644
--- a/src/collision_matrix_model.cpp
+++ b/src/collision_matrix_model.cpp
@@ -42,6 +42,7 @@
 #include <QPalette>
 #include <QApplication>
 #include <QItemSelection>
+#include <QRegularExpression>
 #include <unordered_map>
 
 namespace moveit_setup
@@ -194,10 +195,10 @@ void CollisionMatrixModel::setEnabled(const QModelIndexList& indexes, bool value
     setData(idx, value ? Qt::Checked : Qt::Unchecked, Qt::CheckStateRole);
 }
 
-void CollisionMatrixModel::setFilterRegExp(const QString& filter)
+void CollisionMatrixModel::setFilterRegularExpression(const QString& filter)
 {
   beginResetModel();
-  QRegExp regexp(filter);
+  const QRegularExpression regexp(filter);
   visual_to_index_.clear();
   for (int idx = 0, end = q_names_.size(); idx != end; ++idx)
   {
diff --git a/src/default_collisions_widget.cpp b/src/default_collisions_widget.cpp
index 687378912..b09814513 100644
--- a/src/default_collisions_widget.cpp
+++ b/src/default_collisions_widget.cpp
@@ -275,8 +275,8 @@ void DefaultCollisionsWidget::loadCollisionTable()
     linear_model->setParent(sorted_model);
     matrix_model->setParent(linear_model);
   }
-  connect(link_name_filter_, SIGNAL(textChanged(QString)), model, SLOT(setFilterRegExp(QString)));
-  QMetaObject::invokeMethod(model, "setFilterRegExp", Q_ARG(QString, link_name_filter_->text()));
+  connect(link_name_filter_, SIGNAL(textChanged(QString)), model, SLOT(setFilterRegularExpression(QString)));
+  QMetaObject::invokeMethod(model, "setFilterRegularExpression", Q_ARG(QString, link_name_filter_->text()));
 
   collision_table_->setModel(model);
   // delete old and remember new model
