diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b45187..f24262d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,14 +77,15 @@ set(HEADER_FILES
     )
 
 add_library(${PROJECT_NAME}_gui SHARED ${SOURCE_FILES} ${HEADER_FILES})
-ament_target_dependencies(${PROJECT_NAME}_gui PUBLIC
-  rclcpp
-  rviz_common
-  rviz_rendering
-  rviz_default_plugins
-  rviz_ogre_vendor
+target_link_libraries(${PROJECT_NAME}_gui PUBLIC
+  rclcpp::rclcpp
+  rviz_common::rviz_common
+  rviz_rendering::rviz_rendering
+  rviz_default_plugins::rviz_default_plugins
+  rviz_ogre_vendor::OgreMain
+  rviz_ogre_vendor::OgreOverlay
 )
-target_link_libraries(${PROJECT_NAME}_gui PUBLIC Qt5::Widgets)
+target_link_libraries(${PROJECT_NAME}_gui PUBLIC Qt6::Widgets)
 
 # prevent pluginlib from using boost
 target_compile_definitions(${PROJECT_NAME}_gui PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
@@ -98,18 +99,18 @@ add_library(${PROJECT_NAME}_remote_control SHARED
 # Needed for M_PI on Windows
 target_compile_definitions(${PROJECT_NAME}_remote_control PRIVATE _USE_MATH_DEFINES)
 
-ament_target_dependencies(${PROJECT_NAME}_remote_control
-  rclcpp
-  rclcpp_components
-  visualization_msgs
-  tf2
-  tf2_eigen
-  tf2_geometry_msgs
-  sensor_msgs
-  shape_msgs
-  std_msgs
-  trajectory_msgs
-  eigen_stl_containers
+target_link_libraries(${PROJECT_NAME}_remote_control
+  rclcpp::rclcpp
+  rclcpp_components::component
+  ${visualization_msgs_TARGETS}
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  ${tf2_geometry_msgs_TARGETS}
+  ${sensor_msgs_TARGETS}
+  ${shape_msgs_TARGETS}
+  ${std_msgs_TARGETS}
+  ${trajectory_msgs_TARGETS}
+  eigen_stl_containers::eigen_stl_containers
 )
 
 # Visualization Tools Library
@@ -118,22 +119,22 @@ add_library(${PROJECT_NAME} SHARED
   src/tf_visual_tools.cpp
 )
 target_compile_definitions(${PROJECT_NAME} PRIVATE _USE_MATH_DEFINES)
-ament_target_dependencies(${PROJECT_NAME} PUBLIC Eigen3)
+target_link_libraries(${PROJECT_NAME} Eigen3::Eigen)
 target_link_libraries(${PROJECT_NAME}
   ${PROJECT_NAME}_remote_control
 )
-ament_target_dependencies(${PROJECT_NAME}
-  rclcpp
-  rclcpp_components
-  visualization_msgs
-  tf2
-  tf2_eigen
-  tf2_geometry_msgs
-  sensor_msgs
-  shape_msgs
-  std_msgs
-  trajectory_msgs
-  eigen_stl_containers
+target_link_libraries(${PROJECT_NAME}
+  rclcpp::rclcpp
+  rclcpp_components::component
+  ${visualization_msgs_TARGETS}
+  tf2::tf2
+  tf2_eigen::tf2_eigen
+  ${tf2_geometry_msgs_TARGETS}
+  ${sensor_msgs_TARGETS}
+  ${shape_msgs_TARGETS}
+  ${std_msgs_TARGETS}
+  ${trajectory_msgs_TARGETS}
+  eigen_stl_containers::eigen_stl_containers
 )
 
 # Library
@@ -144,14 +145,14 @@ target_compile_definitions(${PROJECT_NAME}_imarker_simple PRIVATE _USE_MATH_DEFI
 target_link_libraries(${PROJECT_NAME}_imarker_simple
   ${PROJECT_NAME}
 )
-ament_target_dependencies(${PROJECT_NAME}_imarker_simple
-  rclcpp
-  interactive_markers
-  geometry_msgs
-  visualization_msgs
-  sensor_msgs
-  eigen_stl_containers
-  Eigen3
+target_link_libraries(${PROJECT_NAME}_imarker_simple
+  rclcpp::rclcpp
+  interactive_markers::interactive_markers
+  ${geometry_msgs_TARGETS}
+  ${visualization_msgs_TARGETS}
+  ${sensor_msgs_TARGETS}
+  eigen_stl_containers::eigen_stl_containers
+  Eigen3::Eigen
 )
 
 # Demo executable
@@ -162,10 +163,10 @@ target_compile_definitions(${PROJECT_NAME}_demo PRIVATE _USE_MATH_DEFINES)
 target_link_libraries(${PROJECT_NAME}_demo
   ${PROJECT_NAME}
 )
-ament_target_dependencies(${PROJECT_NAME}_demo
-  rclcpp
-  geometry_msgs
-  std_msgs
+target_link_libraries(${PROJECT_NAME}_demo
+  rclcpp::rclcpp
+  ${geometry_msgs_TARGETS}
+  ${std_msgs_TARGETS}
 )
 
 # Demo executable
@@ -175,8 +176,8 @@ target_compile_definitions(${PROJECT_NAME}_imarker_simple_demo PRIVATE _USE_MATH
 target_link_libraries(${PROJECT_NAME}_imarker_simple_demo
   ${PROJECT_NAME}_imarker_simple
 )
-ament_target_dependencies(${PROJECT_NAME}_imarker_simple_demo
-  rclcpp
+target_link_libraries(${PROJECT_NAME}_imarker_simple_demo
+  rclcpp::rclcpp
 )
 
 #############
@@ -248,8 +249,8 @@ if (BUILD_TESTING)
     tests/rvt_test.cpp
     TIMEOUT 180)
   target_compile_definitions(${PROJECT_NAME}_rvt_test PRIVATE _USE_MATH_DEFINES)
-  ament_target_dependencies(${PROJECT_NAME}_rvt_test
-    rclcpp
+  target_link_libraries(${PROJECT_NAME}_rvt_test
+    rclcpp::rclcpp
   )
   target_link_libraries(${PROJECT_NAME}_rvt_test
     ${PROJECT_NAME}
diff --git a/src/rviz_visual_tools.cpp b/src/rviz_visual_tools.cpp
index d4a83bf..da9eda5 100644
--- a/src/rviz_visual_tools.cpp
+++ b/src/rviz_visual_tools.cpp
@@ -42,9 +42,9 @@
 #include <rviz_visual_tools/rviz_visual_tools.hpp>
 
 // Conversions
-#include <tf2/convert.h>
-#include <tf2/LinearMath/Vector3.h>
-#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/convert.hpp>
+#include <tf2/LinearMath/Vector3.hpp>
+#include <tf2/LinearMath/Quaternion.hpp>
 #if __has_include(<tf2_eigen/tf2_eigen.hpp>)
 #include <tf2_eigen/tf2_eigen.hpp>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
diff --git a/src/tf_visual_tools.cpp b/src/tf_visual_tools.cpp
index 99d7aca..acf429a 100644
--- a/src/tf_visual_tools.cpp
+++ b/src/tf_visual_tools.cpp
@@ -39,7 +39,7 @@
 #endif
 
 // TF
-#include <tf2/convert.h>
+#include <tf2/convert.hpp>
 
 // C++
 #include <string>
