diff --git a/mocap4r2_robot_gt/mocap4r2_robot_gt/CMakeLists.txt b/mocap4r2_robot_gt/mocap4r2_robot_gt/CMakeLists.txt
index 7c14e7e..8365d74 100644
--- a/mocap4r2_robot_gt/mocap4r2_robot_gt/CMakeLists.txt
+++ b/mocap4r2_robot_gt/mocap4r2_robot_gt/CMakeLists.txt
@@ -16,13 +16,13 @@ find_package(geometry_msgs REQUIRED)
 find_package(tf2_geometry_msgs REQUIRED)
 
 set(dependencies
-    rclcpp
-    rclcpp_components
-    mocap4r2_msgs
-    mocap4r2_robot_gt_msgs
-    tf2_ros
-    geometry_msgs
-    tf2_geometry_msgs
+    rclcpp::rclcpp
+    rclcpp_components::component
+    ${mocap4r2_msgs_TARGETS}
+    ${mocap4r2_robot_gt_msgs_TARGETS}
+    tf2_ros::tf2_ros
+    ${geometry_msgs_TARGETS}
+    ${tf2_geometry_msgs_TARGETS}
 )
 
 include_directories(
@@ -30,19 +30,19 @@ include_directories(
 )
 
 add_library(gt_component SHARED src/mocap4r2_robot_gt/gt_component.cpp)
-ament_target_dependencies(gt_component ${dependencies})
+target_link_libraries(gt_component ${dependencies})
 rclcpp_components_register_nodes(gt_component "mocap4r2_robot_gt::GTNode")
 
 add_executable(gt_program src/gt_program.cpp)
-ament_target_dependencies(gt_component ${dependencies})
+target_link_libraries(gt_component ${dependencies})
 target_link_libraries(gt_program gt_component)
 
 add_library(set_gt_component SHARED src/mocap4r2_robot_gt/set_gt_component.cpp)
-ament_target_dependencies(set_gt_component ${dependencies})
+target_link_libraries(set_gt_component ${dependencies})
 rclcpp_components_register_nodes(set_gt_component "mocap4r2_robot_gt::SetGTNode")
 
 add_executable(set_gt_cli src/set_gt_cli.cpp)
-ament_target_dependencies(set_gt_cli ${dependencies})
+target_link_libraries(set_gt_cli ${dependencies})
 target_link_libraries(set_gt_cli set_gt_component)
 
 install(TARGETS
