diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6dc113eab..8c8e58b76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,8 +12,7 @@ find_package(Boost REQUIRED program_options)
 find_package(moveit_setup_framework REQUIRED)
 find_package(moveit_setup_srdf_plugins REQUIRED)
 find_package(pluginlib REQUIRED)
-find_package(Qt5Core REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt6 COMPONENTS Core Widgets)
 find_package(rclcpp REQUIRED)
 
 set(THIS_PACKAGE_INCLUDE_DEPENDS
@@ -21,8 +20,8 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
     moveit_setup_framework
     moveit_setup_srdf_plugins
     pluginlib
-    Qt5Core
-    Qt5Widgets
+    Qt6Core
+    Qt6Widgets
     rclcpp)
 
 # Header files that need Qt Moc pre-processing for use with Qt signals, etc:
@@ -32,7 +31,7 @@ set(HEADERS include/moveit_setup_assistant/navigation_widget.hpp
 set(CMAKE_AUTOMOC ON)
 add_definitions(-DQT_NO_KEYWORDS)
 
-qt5_wrap_cpp(MOC_FILES ${HEADERS})
+qt6_wrap_cpp(MOC_FILES ${HEADERS})
 
 add_executable(
   moveit_setup_assistant src/main.cpp src/setup_assistant_widget.cpp
@@ -41,9 +40,16 @@ target_include_directories(
   moveit_setup_assistant
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_setup_assistant>)
-ament_target_dependencies(moveit_setup_assistant
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
-target_link_libraries(moveit_setup_assistant ${Boost_LIBRARIES})
+target_link_libraries(
+  moveit_setup_assistant
+  ${Boost_LIBRARIES}
+  ament_index_cpp::ament_index_cpp
+  moveit_setup_framework::moveit_setup_framework
+  moveit_setup_srdf_plugins::moveit_setup_srdf_plugins
+  pluginlib::pluginlib
+  Qt6::Core
+  Qt6::Widgets
+  rclcpp::rclcpp)
 
 add_executable(moveit_setup_assistant_updater src/collisions_updater.cpp)
 target_include_directories(
@@ -51,8 +57,16 @@ target_include_directories(
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
          $<INSTALL_INTERFACE:include/moveit_setup_assistant>)
 
-ament_target_dependencies(moveit_setup_assistant_updater
-                          ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(
+  moveit_setup_assistant_updater
+  ${Boost_LIBRARIES}
+  ament_index_cpp::ament_index_cpp
+  moveit_setup_framework::moveit_setup_framework
+  moveit_setup_srdf_plugins::moveit_setup_srdf_plugins
+  pluginlib::pluginlib
+  Qt6::Core
+  Qt6::Widgets
+  rclcpp::rclcpp)
 
 set_target_properties(moveit_setup_assistant_updater
                       PROPERTIES OUTPUT_NAME collisions_updater PREFIX "")
